detect/detect.go

```diff
@@ -322,20 +368,28 @@ func (d *Detector) detectRule(fragment Fragment, currentRaw string, r config.Rul
 			// Path _only_ rule
 			if r.Path.MatchString(fragment.FilePath) || (fragment.WindowsFilePath != "" && r.Path.MatchString(fragment.WindowsFilePath)) {
 				finding := report.Finding{
+					Commit:      fragment.CommitSHA,
 					RuleID:      r.RuleID,
 					Description: r.Description,
 					File:        fragment.FilePath,
 					SymlinkFile: fragment.SymlinkFile,
-					Match:       fmt.Sprintf("file detected: %s", fragment.FilePath),
+					Match:       "file detected: " + fragment.FilePath,
 					Tags:        r.Tags,
 				}
+				if fragment.CommitInfo != nil {
+					finding.Author = fragment.CommitInfo.AuthorName
+					finding.Date = fragment.CommitInfo.Date
+					finding.Email = fragment.CommitInfo.AuthorEmail
+					finding.Link = createScmLink(fragment.CommitInfo.Remote, finding)
+					finding.Message = fragment.CommitInfo.Message
+				}
 				return append(findings, finding)
 			}
 		} else {
 			// if path is set _and_ a regex is set, then we need to check both
 			// so if the path does not match, then we should return early and not
 			// consider the regex
-			if !(r.Path.MatchString(fragment.FilePath) || (fragment.WindowsFilePath != "" && r.Path.MatchString(fragment.WindowsFilePath))) {
+			if !r.Path.MatchString(fragment.FilePath) && (fragment.WindowsFilePath == "" || !r.Path.MatchString(fragment.WindowsFilePath)) {
```

Comment:
Was this a De Morgan's Law suggestion? I think this is significantly harder to read than the prior version.

Reply:
Yeah. It was staticcheck QF1001. 

To me the newer version is slightly easier to keep in my head[1], but I'll go ahead and revert and exclude `QF1001` in the golangci-lint to keep things how they were. :ok_hand: 

[1] The `!(...)` means I just have to keep a little more in my head, but it also reads more like normal English I think.